temporal/server_test.go

```diff
@@ -98,35 +101,84 @@ func setTestPorts(cfg *config.Config) {
 	}
 }
 
+type errorLogDetector struct {
+	t  testing.TB
+	on atomic.Bool
+	log.Logger
+}
+
+func (d *errorLogDetector) Start() {
+	d.on.Store(true)
+}
+
+func (d *errorLogDetector) Stop() {
+	d.on.Store(false)
+}
+
+func (d *errorLogDetector) Warn(msg string, _ ...tag.Tag) {
```

Comment:
should this (and Error) pass through the message to the underlying Logger? if not, maybe we shouldn't even have the Logger and just implement the rest of the methods as no-ops? if so, maybe we should make the Logger a stdout logger for help debugging?

Reply:
The underlying logger was a `noopLogger`, and I re-used it for the sake of brevity. However, I think that using a real logger and delegating to it for debugging is probably better. Updated.